src/twisted/web/client.py

```diff
@@ -1714,6 +1714,9 @@ def request(self, method, uri, headers=None, bodyProducer=None):
 
         @see: L{twisted.web.iweb.IAgent.request}
         """
+        if not isinstance(method, bytes):
```

Comment:
Wouldn't it be better to do it in `_AgentBase._requestWithEndpoint`, so that `ProxyAgent` won't have literally the same bug?

This is not a blocker, but should be considered.

Reply:
If it's not a blocker, a new ticket for that issue should at least be filed.

Reply:
Why not make it a blocker, and also fix the ProxyAgent ?

Reply:
At this point, probably because moshez already approved the PR and it is contributor unfriendly to have conflicting requirements imposed on their contributions.  If there's an issue here, the reasonable next step is probably to discuss with moshez why he didn't think this should be a blocker and come to some agreement for future work.

FWIW, the PR as-is seems like a specific, concrete improvement to a feature of Twisted.  It is not incomplete because it does not also address `ProxyAgent`.  It would be great to *also* fix `ProxyAgent` but no harm is done by taking on that work separately.

Reply:
Thanks. I agree.